### ISSUE TITLE
Fix host/device compilation warnings 

### ISSUE THREAD

**Issue by mparno** [MeasureTransport/MParT#397]
During compilation with CUDA, I'm getting a bunch of warnings that look like the following and should be addressed.
```bash
MParT/BasisEvaluator.h(321): warning #20011-D: calling a __host__ function("std::vector<    ::std::shared_ptr<    ::TestEvaluators> , ::std::allocator<    ::std::shared_ptr<    ::TestEvaluators> > > ::operator [](unsigned long) const") from a __host__ __device__ function("mpart::BasisEvaluator<( ::mpart::BasisHomogeneity)2,     ::std::vector<    ::std::shared_ptr<    ::TestEvaluators> , ::std::allocator<    ::std::shared_ptr<    ::TestEvaluators> > > ,  ::mpart::Identity> ::EvaluateSecondDerivatives const") is not allowed
```

**Comment by dannys4** [comment]
Related: #367

**Comment by dannys4** [comment]
Fixed by #418
